.circleci/config.yml

```diff
@@ -53,9 +53,21 @@ jobs:
       - install_rust
       - install_node_devlibs
       - print_current_versions
+      - run:
+          name: "Compile openssl static libs"
+          command: |
+            set -v
+            cd ~/ && curl -O https://www.openssl.org/source/openssl-1.1.1c.tar.gz
```

Comment:
Is there a way of how we can make this a more obious dependency?
Having to remember that we need to update this version string here regularly seems like a PITA :(
What are we even using openssl for in cnd/btsieve? Can we maybe feature flag that away so that we don't even depend on this?

Reply:
> Is there a way of how we can make this a more obvious dependency?
What do you mean?

> Having to remember that we need to update this version string here regularly seems like a PITA :(
True but I don't see how we can do better

> What are we even using openssl for in cnd/btsieve? Can we maybe feature flag that away so that we don't even depend on this?

There is actually a `vendored` feature flag on `rust-ssl` crate that would give the same result (as far as I understand) and make all this stuff easier.

I quickly check why we depend on ssl when doing this issue:
```
openssl < native-tls < reqwest*
                     < tokio-tls < websocket < web3*
                     < web3*
                     < websocket < web3*
                     < hyper-tls < reqwest*
                                 < web3*
```

So in short, it's brought it by `reqwest` and `web3` crate.

## reqwest
Reqwest has a [`default-tls-vendored` feature](https://github.com/seanmonstar/reqwest/blob/87a09322d6fef48e7a7c79a854acaf08bb16f606/Cargo.toml#L78) that could help.
Also, all tls crates are optional so maybe we could fully remove tls by setting features to `[]`.

## web3
No `vendored` option but the [TLS crates are optionals](https://github.com/tomusdrw/rust-web3/blob/db4443ee16765c0754a87d9359d8f6b9e0c04d3a/Cargo.toml#L45) so same than reqwest.

---

trying with `features = []` for `web3` and `reqwest` now.

Reply:
I am not it is possible to go down the features rabbit hole due to the dependency on testcontainers that also bring tls in.

Reply:
Please note that because we use ubuntu LTS in our CI, we don't even use the latest openssl library (1.1) as only 1.0 is available on ubuntu 16.04.

Hence, I'd be keen to have a follow-up ticket to (try) remove the openssl dependency.

In any case, we do not really on openssl functionally so I am not worried about upgrading openssl version.

What I am worried about is that if someone download our binary right now, it is likely to not work without them messing around symlink in their `/lib` (and installing libzmq), meaning our binary release is pretty much useless as it would actually be more straightforward to compile from source than fix the dynamic lib error they'd have.

Reply:
Ok, the features path lead nowhere. Most likely because test containers also bring openssl in.

This is the best I can do in a reasonable amount of time.

Reply:
Nice research, thanks! :)
Agree on the timing problem! Let's fix it fast and have an issue that tracks how to fix it properly (by removing ssl dependency).
I am curious on why testcontainers brings in SSL, will investigate! :)

Reply:
testcontainers bring in SSL because of web3 and also because of our bitcoin_rpc_client which bring in json_client that brings in tls and hence ssl.